diff --git a/libgnomekbd/gkbd-keyboard-drawing.c b/libgnomekbd/gkbd-keyboard-drawing.c
index 01d8281..9afa7fa 100644
--- a/libgnomekbd/gkbd-keyboard-drawing.c
+++ b/libgnomekbd/gkbd-keyboard-drawing.c
@@ -1305,7 +1305,7 @@ create_cairo (GkbdKeyboardDrawing * drawing)
 	if (drawing == NULL || drawing->surface == NULL)
 		return FALSE;
 	drawing->renderContext->cr =
-	    gdk_cairo_create (GDK_DRAWABLE (drawing->surface));
+	    gdk_cairo_create (GDK_WINDOW (drawing->surface));
 
 	state = gtk_widget_get_state (GTK_WIDGET (drawing));
 	drawing->renderContext->dark_color =
