--- revoco.c	2010-01-03 17:37:53.011597016 +0100
+++ revoco.c	2010-01-03 17:39:36.091627832 +0100
@@ -4,6 +4,7 @@
  * Requires hiddev.
  *
  * Written November 2006 by E. Toernig's bonobo - no copyrights.
+ * Cleanup by Petteri Räty <betelgeuse@gentoo.org> and adapted to 0.5 by Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
  *
  * Contact: Edgar Toernig <froese@gmx.de>
  *
@@ -54,6 +55,9 @@
 #include <unistd.h>
 #include <sys/ioctl.h>
 #include <sys/select.h>
+// hiddev and the headers it requires
+#include <asm/types.h>
+#include <linux/hiddev.h>
 
 #define streq(a,b)	(strcmp((a), (b)) == 0)
 #define strneq(a,b,c)	(strncmp((a), (b), (c)) == 0)
@@ -65,64 +69,6 @@
 
 static int first_byte;
 
-/*** extracted from hiddev.h ***/
-
-typedef signed short s16;
-typedef signed int s32;
-typedef unsigned int u32;
-
-struct hiddev_devinfo {
-	u32 bustype;
-	u32 busnum;
-	u32 devnum;
-	u32 ifnum;
-	s16 vendor;
-	s16 product;
-	s16 version;
-	u32 num_applications;
-};
-
-struct hiddev_report_info {
-	u32 report_type;
-	u32 report_id;
-	u32 num_fields;
-};
-
-#define HID_REPORT_TYPE_INPUT	1
-#define HID_REPORT_TYPE_OUTPUT	2
-#define HID_REPORT_TYPE_FEATURE	3
-
-struct hiddev_usage_ref {
-	u32 report_type;
-	u32 report_id;
-	u32 field_index;
-	u32 usage_index;
-	u32 usage_code;
-	s32 value;
-};
-
-struct hiddev_usage_ref_multi {
-	struct hiddev_usage_ref uref;
-	u32 num_values;
-	s32 values[1024];
-};
-
-#define HIDIOCGDEVINFO		_IOR('H', 0x03, struct hiddev_devinfo)
-#define HIDIOCINITREPORT	_IO('H', 0x05)
-#define HIDIOCGREPORT		_IOW('H', 0x07, struct hiddev_report_info)
-#define HIDIOCSREPORT		_IOW('H', 0x08, struct hiddev_report_info)
-#define HIDIOCGUSAGES		_IOWR('H', 0x13, struct hiddev_usage_ref_multi)
-#define HIDIOCSUSAGES		_IOW('H', 0x14, struct hiddev_usage_ref_multi)
-#define HIDIOCGFLAG		_IOR('H', 0x0E, int)
-#define HIDIOCSFLAG		_IOW('H', 0x0F, int)
-
-#define HIDDEV_FLAG_UREF	0x1
-#define HIDDEV_FLAG_REPORT	0x2
-#define HIDDEV_FLAGS		0x3
-
-/*** end hiddev.h ***/
-
-
 static void
 fatal(const char *fmt, ...)
 {
@@ -486,8 +432,8 @@
 static void
 usage(void)
 {
-    printf("Revoco v"VERSION" - Change the wheel behaviour of "
-				    "Logitech's MX-Revolution mouse.\n\n");
+    printf("Revoco v%.1f - Change the wheel behaviour of "
+				    "Logitech's MX-Revolution mouse.\n\n", VERSION);
     printf("Usage:\n");
     printf("  revoco free                      free spinning mode\n");
     printf("  revoco click                     click-to-click mode\n");
