diff --git a/js/ui/statusIconDispatcher.js b/js/ui/statusIconDispatcher.js
index c48f276..ecbbc56 100644
--- a/js/ui/statusIconDispatcher.js
+++ b/js/ui/statusIconDispatcher.js
@@ -9,6 +9,8 @@ const NotificationDaemon = imports.ui.notificationDaemon;
 const Util = imports.misc.util;
 
 const STANDARD_TRAY_ICON_IMPLEMENTATIONS = {
+    'parcellite': 'parcellite',
+    'dropbox': 'dropbox',
     'bluetooth-applet': 'bluetooth',
     'gnome-volume-control-applet': 'volume', // renamed to gnome-sound-applet
                                              // when moved to control center
