From 719e337b0235f6843d34bf50c0cd88988bf2c198 Mon Sep 17 00:00:00 2001
From: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Date: Tue, 15 Feb 2011 21:06:36 +0100
Subject: [PATCH 2/2] whitelist dropbox & gnote

---
 js/ui/statusIconDispatcher.js |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/js/ui/statusIconDispatcher.js b/js/ui/statusIconDispatcher.js
index da1535c..a590a89 100644
--- a/js/ui/statusIconDispatcher.js
+++ b/js/ui/statusIconDispatcher.js
@@ -10,6 +10,8 @@ const Util = imports.misc.util;
 
 const STANDARD_TRAY_ICON_IMPLEMENTATIONS = {
     'gpaste-applet': 'gpaste-applet',
+    'dropbox': 'dropbox',
+    'gnote': 'gnote',
     'bluetooth-applet': 'bluetooth',
     'gnome-volume-control-applet': 'volume', // renamed to gnome-sound-applet
                                              // when moved to control center
-- 
1.7.4.1.26.g00e6e.dirty

