--- js/ui/panel.js.old	2010-03-05 18:55:49.617792160 +0100
+++ js/ui/panel.js	2010-03-05 18:56:54.505793440 +0100
@@ -610,11 +610,10 @@
 
     _updateClock: function() {
         let displayDate = new Date();
-        let msecRemaining = 60000 - (1000 * displayDate.getSeconds() +
-                                     displayDate.getMilliseconds());
+        let msecRemaining = 1000 - displayDate.getMilliseconds();
         if (msecRemaining < 500) {
-            displayDate.setMinutes(displayDate.getMinutes() + 1);
-            msecRemaining += 60000;
+            displayDate.setSeconds(displayDate.getSeconds() + 1);
+            msecRemaining += 1000;
         }
 
         // if the locale representations of 05:00 and 17:00 do not
@@ -627,10 +626,10 @@
                         fivePm.toLocaleFormat("%X").substr(0,2);
         if (isTime24h) {
             /* Translators: This is the time format used in 24-hour mode. */
-            this._clock.set_text(displayDate.toLocaleFormat(_("%a %R")));
+            this._clock.set_text(displayDate.toLocaleFormat(_("%a %R:%S")));
         } else {
             /* Translators: This is a time format used for AM/PM. */
-            this._clock.set_text(displayDate.toLocaleFormat(_("%a %l:%M %p")));
+            this._clock.set_text(displayDate.toLocaleFormat(_("%a %l:%M:%S %p")));
         }
         Mainloop.timeout_add(msecRemaining, Lang.bind(this, this._updateClock));
         return false;
