diff --git a/src/contact-channel.c b/src/contact-channel.c
index d87a81a..339f02c 100644
--- a/src/contact-channel.c
+++ b/src/contact-channel.c
@@ -225,8 +225,6 @@ salut_contact_channel_set_property (GObject     *object,
       priv->conn = g_value_get_object (value);
       break;
     case PROP_HANDLE_TYPE:
-      g_assert (g_value_get_uint (value) == 0
-               || g_value_get_uint (value) == TP_HANDLE_TYPE_LIST);
       break;
     case PROP_CHANNEL_TYPE:
       tmp = g_value_get_string (value);
diff --git a/src/file-transfer-channel.c b/src/file-transfer-channel.c
index 794631f..89dc884 100644
--- a/src/file-transfer-channel.c
+++ b/src/file-transfer-channel.c
@@ -354,8 +354,6 @@ salut_file_transfer_channel_set_property (GObject *object,
         self->priv->connection = g_value_get_object (value);
         break;
       case PROP_HANDLE_TYPE:
-        g_assert (g_value_get_uint (value) == 0
-                  || g_value_get_uint (value) == TP_HANDLE_TYPE_CONTACT);
         break;
       case PROP_CHANNEL_TYPE:
         /* these properties are writable in the interface, but not actually
diff --git a/src/im-channel.c b/src/im-channel.c
index 4843e38..816a58f 100644
--- a/src/im-channel.c
+++ b/src/im-channel.c
@@ -250,8 +250,6 @@ salut_im_channel_set_property (GObject *object,
         priv->connection = g_value_get_object (value);
         break;
       case PROP_HANDLE_TYPE:
-        g_assert (g_value_get_uint (value) == 0 ||
-            g_value_get_uint (value) == TP_HANDLE_TYPE_CONTACT);
         break;
       case PROP_CHANNEL_TYPE:
         tmp = g_value_get_string (value);
diff --git a/src/muc-channel.c b/src/muc-channel.c
index b8b0fbb..18dd749 100644
--- a/src/muc-channel.c
+++ b/src/muc-channel.c
@@ -268,8 +268,6 @@ salut_muc_channel_set_property (GObject     *object,
       priv->muc_connection = g_value_get_object (value);
       break;
     case PROP_HANDLE_TYPE:
-      g_assert (g_value_get_uint (value) == 0
-               || g_value_get_uint (value) == TP_HANDLE_TYPE_ROOM);
       break;
     case PROP_CHANNEL_TYPE:
       tmp = g_value_get_string (value);
