--- clutter-gtk/Makefile.in.old	2010-10-03 13:09:46.226617516 +0200
+++ clutter-gtk/Makefile.in	2010-10-03 13:10:27.700367873 +0200
@@ -744,7 +744,7 @@
 @HAVE_INTROSPECTION_TRUE@	$(AM_V_GEN)$(INTROSPECTION_SCANNER) -v \
 @HAVE_INTROSPECTION_TRUE@		--namespace GtkClutter --nsversion=@CLUTTER_GTK_API_VERSION@ \
 @HAVE_INTROSPECTION_TRUE@		--strip-prefix=GtkClutter \
-@HAVE_INTROSPECTION_TRUE@		$(INCLUDES) $(AM_CPPFLAGS) \
+@HAVE_INTROSPECTION_TRUE@		$(INCLUDES) $(GTK_CFLAGS) $(AM_CPPFLAGS) \
 @HAVE_INTROSPECTION_TRUE@		--add-include-path=$(srcdir) --add-include=path=. \
 @HAVE_INTROSPECTION_TRUE@		--include=Clutter-1.0 \
 @HAVE_INTROSPECTION_TRUE@		--include=GdkPixbuf-2.0 \
