--- src/empathy-status-icon.c	2010-01-12 23:49:32.310135631 +0100
+++ src/empathy-status-icon.c	2010-01-12 23:51:50.415139341 +0100
@@ -154,21 +154,17 @@
 
 	if (priv->event) {
 		gchar *message_esc = NULL;
-		gchar *header_esc = NULL;
 
 		if (priv->event->message != NULL)
 			message_esc = g_markup_escape_text (priv->event->message, -1);
 
-		if (priv->event->header != NULL)
-			header_esc = g_markup_escape_text (priv->event->header, -1);
-
 		if (priv->notification) {
 			notify_notification_update (priv->notification,
-						    header_esc, message_esc,
+						    priv->event->header, message_esc,
 						    NULL);
 		} else {
 			priv->notification = notify_notification_new_with_status_icon
-				(header_esc, message_esc, NULL, priv->icon);
+				(priv->event->header, message_esc, NULL, priv->icon);
 			notify_notification_set_timeout (priv->notification,
 							 NOTIFY_EXPIRES_DEFAULT);
 
@@ -199,7 +195,6 @@
 		notify_notification_show (priv->notification, NULL);
 
 		g_free (message_esc);
-		g_free (header_esc);
 	} else {
 		notification_close_helper (priv);
 	}
