--- configure.ac.old	2010-02-05 15:27:08.309784965 +0100
+++ configure.ac	2010-02-05 15:27:34.173857004 +0100
@@ -45,7 +45,6 @@
 PANGO_REQUIRED_VERSION=1.3.0
 SCROLLKEEPER_REQUIRED_VERSION=0.1.4
 GCONF_REQUIRED_VERSION=2.31.3
-GNOME_PANEL_REQUIRED_VERSION=2.0.0
 LIBXKLAVIER_REQUIRED_VERSION=4.0
 LIBCANBERRA_GTK_REQUIRED_VERSION=0.4
 #FONTCONFIG_REQUIRED_VERSION=2.6.0
@@ -149,13 +148,6 @@
 AC_SUBST(SIMPLE_CHOOSER_CFLAGS)
 AC_SUBST(SIMPLE_CHOOSER_LIBS)
 
-PKG_CHECK_MODULES(APPLET,
-        dbus-glib-1 >= $DBUS_GLIB_REQUIRED_VERSION
-        gtk+-2.0 >= $GTK_REQUIRED_VERSION
-        libpanelapplet-2.0 >= $GNOME_PANEL_REQUIRED_VERSION)
-AC_SUBST(APPLET_CFLAGS)
-AC_SUBST(APPLET_LIBS)
-
 PLUGIN_LIBTOOL_FLAGS="-export_dynamic -module -avoid-version"
 AC_SUBST(PLUGIN_LIBTOOL_FLAGS)
 
@@ -1402,7 +1395,6 @@
 gui/simple-greeter/Makefile
 gui/simple-greeter/libnotificationarea/Makefile
 gui/simple-chooser/Makefile
-gui/user-switch-applet/Makefile
 utils/Makefile
 data/gdm.conf
 data/Makefile

--- gui/Makefile.am.old	2010-08-19 14:25:40.204000018 +0200
+++ gui/Makefile.am	2010-08-19 14:25:52.676000013 +0200
@@ -2,7 +2,6 @@
 
 SUBDIRS =			\
 	simple-greeter		\
-	user-switch-applet	\
 	$(NULL)
 
 if XDMCP_SUPPORT
@@ -12,5 +11,4 @@
 DIST_SUBDIRS =			\
 	simple-chooser		\
 	simple-greeter		\
-	user-switch-applet	\
 	$(NULL)
