--- a/libbrasero-utils/brasero-jacket-buffer.c	2010-09-06 14:48:42.260779817 +0200
+++ b/libbrasero-utils/brasero-jacket-buffer.c	2010-09-06 14:58:11.959598434 +0200
@@ -76,7 +76,7 @@
 	guint left_margin_accumulative = 0;
 	guint right_margin_accumulative = 0;
 
-	g_return_if_fail (!dest->realized);
+	g_return_if_fail (!dest);
 
 	for (; tags; tags = tags->next) {
 		GtkTextTag *tag;
@@ -162,17 +162,13 @@
 		}
 
 		if (bg_stipple_set) {
-			if (dest->appearance.bg_stipple)
-				g_object_unref (dest->appearance.bg_stipple);
-			g_object_get (tag, "background-stipple", &dest->appearance.bg_stipple, NULL);
+			g_object_get (tag, "background-stipple", NULL, NULL);
 
 			dest->appearance.draw_bg = TRUE;
 		}
 
 		if (fg_stipple_set) {
-			if (dest->appearance.fg_stipple)
-				g_object_unref (dest->appearance.fg_stipple);
-			g_object_get (tag, "foreground-stipple", &dest->appearance.fg_stipple, NULL);
+			g_object_get (tag, "foreground-stipple", NULL, NULL);
 		}
 
 		if (font_desc) {
