--- library/gnome-keyring.c.old	2009-12-14 03:34:45.000000000 +0100
+++ library/gnome-keyring.c	2010-03-06 12:59:47.272186879 +0100
@@ -4333,11 +4333,20 @@
 
 static void
 find_password_1_reply (GkrOperation *op, const char *path, gpointer user_data)
 {
+
+	GkrCallback *cb;
+
+	/* All done, complete the operation here */
+	if (path == NULL) {
+		cb = gkr_operation_pop (op);
+		gkr_callback_invoke_ok_string (cb, NULL);
 	/* We need a session to get the secret for this item */
-	gkr_operation_push (op, find_password_2_reply, GKR_CALLBACK_OP_SESSION,
-	                    g_strdup (path), g_free);
-	gkr_session_negotiate (op);
+	} else {
+		gkr_operation_push (op, find_password_2_reply, GKR_CALLBACK_OP_SESSION,
+	        	            g_strdup (path), g_free);
+		gkr_session_negotiate (op);
+	}
 }
 
 static GkrOperation*
