From dc457ad59773bc60e6d5f9414447b76c6a20fb48 Mon Sep 17 00:00:00 2001
From: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Date: Mon, 12 Dec 2011 14:38:16 +0100
Subject: [PATCH 2/2] Disable wacom

Signed-off-by: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
---
 configure.ac        |    6 ------
 plugins/Makefile.am |    1 -
 2 files changed, 0 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5b8e2af..a38add6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -221,12 +221,6 @@ dnl ---------------------------------------------------------------------------
 
 PKG_CHECK_MODULES(MEDIA_KEYS, [gio-unix-2.0 libpulse >= $PA_REQUIRED_VERSION $GUDEV_PKG libpulse-mainloop-glib >= $PA_REQUIRED_VERSION libcanberra])
 
-dnl ---------------------------------------------------------------------------
-dnl - wacom
-dnl ---------------------------------------------------------------------------
-
-PKG_CHECK_MODULES(WACOM, [libwacom >= $LIBWACOM_REQUIRED_VERSION x11 xi gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION])
-
 dnl ==============================================
 dnl PackageKit section
 dnl ==============================================
diff --git a/plugins/Makefile.am b/plugins/Makefile.am
index 31fcb74..e40d9b9 100644
--- a/plugins/Makefile.am
+++ b/plugins/Makefile.am
@@ -14,7 +14,6 @@ enabled_plugins =	\
 	media-keys	\
 	mouse		\
 	sound		\
-	wacom		\
 	xrandr		\
 	xsettings	\
 	$(NULL)
-- 
1.7.9.106.g2a2aa.dirty

