From 01e9a56df9abbd9dacccb215969e2dc1d46366d7 Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Thu, 26 Aug 2010 14:08:26 +0000
Subject: Bug 29816 – Install polkitagentenumtypes.h

https://bugs.freedesktop.org/show_bug.cgi?id=29816

Signed-off-by: David Zeuthen <davidz@redhat.com>
---
diff --git a/src/polkitagent/Makefile.am b/src/polkitagent/Makefile.am
index d2fc9c8..96ffd7c 100644
--- a/src/polkitagent/Makefile.am
+++ b/src/polkitagent/Makefile.am
@@ -46,6 +46,7 @@ libpolkit_agent_1includedir=$(includedir)/polkit-1/polkitagent
 
 libpolkit_agent_1include_HEADERS =                        				\
 	polkitagent.h									\
+	polkitagentenumtypes.h								\
 	polkitagenttypes.h								\
 	polkitagentsession.h								\
 	polkitagentlistener.h								\
--
cgit v0.8.3-6-g21f6
