diff --git a/champlain/Makefile.am b/champlain/Makefile.am
index bdaa33e..d77a59e 100644
--- a/champlain/Makefile.am
+++ b/champlain/Makefile.am
@@ -110,7 +110,7 @@ libchamplain_include_HEADERS = \
 	champlain-version.h		\
 	champlain-features.h		
 
-libchamplain_@CHAMPLAIN_API_VERSION@_la_LIBADD = $(DEPS_LIBS) $(SOUP_LIBS) $(MEMPHIS_LIBS) ../tidy/libtidy-1.0.la
+libchamplain_@CHAMPLAIN_API_VERSION@_la_LIBADD = $(DEPS_LIBS) $(SOUP_LIBS) $(SOUP_GNOME_LIBS) $(MEMPHIS_LIBS) ../tidy/libtidy-1.0.la
 
 libchamplain_includedir = $(includedir)/libchamplain-@CHAMPLAIN_API_VERSION@/champlain
 
@@ -122,6 +122,7 @@ libchamplain_@CHAMPLAIN_API_VERSION@_la_LDFLAGS = -version-info $(LIBRARY_VERSIO
 AM_CPPFLAGS = 				\
 	$(DEPS_CFLAGS)			\
 	$(SOUP_CFLAGS)			\
+	$(SOUP_GNOME_CFLAGS)    \
 	$(MEMPHIS_CFLAGS)		\
 	-DDATADIR=\""$(datadir)"\"	\
 	-I$(top_srcdir)/tidy		\
diff --git a/demos/Makefile.am b/demos/Makefile.am
index d095960..8df00d9 100644
--- a/demos/Makefile.am
+++ b/demos/Makefile.am
@@ -15,8 +15,8 @@ polygons_SOURCES = polygons.c
 polygons_LDADD = $(DEPS_LIBS) ../champlain/libchamplain-@CHAMPLAIN_API_VERSION@.la
 
 url_marker_SOURCES = url-marker.c
-url_marker_CPPFLAGS = $(DEPS_CFLAGS) $(SOUP_CFLAGS)
-url_marker_LDADD = $(SOUP_LIBS) $(DEPS_LIBS) ../champlain/libchamplain-@CHAMPLAIN_API_VERSION@.la
+url_marker_CPPFLAGS = $(DEPS_CFLAGS) $(SOUP_CFLAGS) $(SOUP_GNOME_CFLAGS)
+url_marker_LDADD = $(SOUP_LIBS) $(DEPS_LIBS) $(SOUP_GNOME_LIBS) ../champlain/libchamplain-@CHAMPLAIN_API_VERSION@.la
 
 if ENABLE_GTK
 noinst_PROGRAMS += launcher-gtk
