--- gjs-0.4/modules/console.c.old	2009-06-05 18:50:19.000000000 +0200
+++ gjs-0.4/modules/console.c	2009-11-14 12:32:55.907649842 +0100
@@ -50,7 +50,6 @@
 #endif
 
 #include <jsapi.h>
-#include <jsstddef.h> /* PTRDIFF */
 #include <glib.h>
 #include <glib/gprintf.h>
 #include <gjs/gjs.h>
@@ -111,7 +110,7 @@
             report->linebuf,
             (n > 0 && report->linebuf[n-1] == '\n') ? "" : "\n",
             prefix);
-    n = PTRDIFF(report->tokenptr, report->linebuf, char);
+    n = (((unsigned int) report->tokenptr) - ((unsigned int) report->linebuf))/sizeof(char);
     for (i = j = 0; i < n; i++) {
         if (report->linebuf[i] == '\t') {
             for (k = (j + 8) & ~7; j < k; j++) {
--- gjs-0.4/gjs/jsapi-util.c.old	2009-08-24 16:54:11.000000000 +0200
+++ gjs-0.4/gjs/jsapi-util.c	2009-11-14 12:42:20.907637322 +0100
@@ -32,7 +32,6 @@
 #include "context-jsapi.h"
 
 #include <string.h>
-#include <jscntxt.h>
 
 GQuark 
 gjs_util_error_quark (void)
--- gjs-0.4/gjs/jsapi-util.hi.old	2009-08-14 18:45:58.000000000 +0200
+++ gjs-0.4/gjs/jsapi-util.h	2009-11-14 12:46:18.251646808 +0100
@@ -31,6 +31,14 @@
 #include <jsapi.h>
 #include <glib-object.h>
 
+typedef enum JSErrNum {
+#define MSG_DEF(name, number, count, exception, format) \
+    name = number,
+MSG_DEF(JSMSG_OUT_OF_MEMORY,          137, 0, JSEXN_ERR, "out of memory")
+#undef MSG_DEF
+    JSErr_Limit
+} JSErrNum;
+
 G_BEGIN_DECLS
 
 #define GJS_UTIL_ERROR gjs_util_error_quark ()
