From e4a74d72156e4027edc397700b683bedee485bce Mon Sep 17 00:00:00 2001
From: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Date: Thu, 16 Sep 2010 23:28:55 +0200
Subject: [PATCH] Fix introspection build order

---
 clutter/Makefile.am |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/clutter/Makefile.am b/clutter/Makefile.am
index f601bd5..2372e2d 100644
--- a/clutter/Makefile.am
+++ b/clutter/Makefile.am
@@ -368,6 +368,9 @@ Clutter_@CLUTTER_API_VERSION_AM@_gir_SCANNERFLAGS = \
 	$(json_gir_include) \
 	--include-uninstalled=$(top_builddir)/clutter/cogl/cogl/Cogl-@CLUTTER_API_VERSION@.gir
 
+# Clutter.gir includes ClutterJson.gir so it needs to be generated before
+Clutter-@CLUTTER_API_VERSION@.gir: $(clutter_json_gir)
+
 # ClutterJson.gir and Cogl.gir are used included into Clutter.gir, so they
 # need to be built before the typelib is generated
 Clutter-@CLUTTER_API_VERSION@.typelib: $(clutter_json_gir) Cogl-@CLUTTER_API_VERSION@.gir
-- 
1.7.2.3

