From 443d48907374d5f2e34fc492448e35d028428faa Mon Sep 17 00:00:00 2001
From: Javier Jardón <jjardon@gnome.org>
Date: Wed, 08 Jun 2011 17:29:52 +0000
Subject: docs: Use const instead G_CONST_RETURN

https://bugzilla.gnome.org/show_bug.cgi?id=652129
---
diff --git a/doc/CODING_STYLE b/doc/CODING_STYLE
index 9f0a15b..b8682c8 100644
--- a/doc/CODING_STYLE
+++ b/doc/CODING_STYLE
@@ -406,10 +406,10 @@ definition should be vertically aligned in three columns:
 The maximum width of each column is given by the longest element in the
 column:
 
-  void                  clutter_type_set_property (ClutterType  *type,
-                                                   const gchar  *value,
-                                                   GError      **error);
-  G_CONST_RETURN gchar *clutter_type_get_property (ClutterType  *type);
+  void         clutter_type_set_property (ClutterType  *type,
+                                          const gchar  *value,
+                                          GError      **error);
+  const gchar *clutter_type_get_property (ClutterType  *type);
 
 It is also possible to align the columns to the next tab:
 
diff --git a/doc/cookbook/examples/cb-button.c b/doc/cookbook/examples/cb-button.c
index 0260d14..e02416c 100644
--- a/doc/cookbook/examples/cb-button.c
+++ b/doc/cookbook/examples/cb-button.c
@@ -429,7 +429,7 @@ cb_button_set_text_color (CbButton           *self,
  *
  * Returns: the button's text. This must not be freed by the application.
  */
-G_CONST_RETURN gchar *
+const gchar *
 cb_button_get_text (CbButton *self)
 {
   g_return_val_if_fail (CB_IS_BUTTON (self), NULL);
diff --git a/doc/cookbook/examples/cb-button.h b/doc/cookbook/examples/cb-button.h
index 9362a49..cf8adaf 100644
--- a/doc/cookbook/examples/cb-button.h
+++ b/doc/cookbook/examples/cb-button.h
@@ -67,7 +67,7 @@ struct _CbButtonClass
 ClutterActor *cb_button_new (void);
 
 /* getter */
-G_CONST_RETURN gchar * cb_button_get_text (CbButton *self);
+const gchar * cb_button_get_text (CbButton *self);
 
 /* setters - these are wrappers round functions
  * which change properties of the internal actors
--
cgit v0.9
