From a8af207ddcf535f9b1d0a2e99999a9d68b06ea60 Mon Sep 17 00:00:00 2001
From: Javier Jardón <jjardon@gnome.org>
Date: Wed, 08 Jun 2011 17:45:52 +0000
Subject: clutter: Use const instead G_CONST_RETURN

https://bugzilla.gnome.org/show_bug.cgi?id=652129
---
diff --git a/clutter/clutter-actor-meta-private.h b/clutter/clutter-actor-meta-private.h
index e97d8f1..02396eb 100644
--- a/clutter/clutter-actor-meta-private.h
+++ b/clutter/clutter-actor-meta-private.h
@@ -56,7 +56,7 @@ void                  _clutter_meta_group_add_meta    (ClutterMetaGroup *group,
                                                        ClutterActorMeta *meta);
 void                  _clutter_meta_group_remove_meta (ClutterMetaGroup *group,
                                                        ClutterActorMeta *meta);
-G_CONST_RETURN GList *_clutter_meta_group_peek_metas  (ClutterMetaGroup *group);
+const GList *         _clutter_meta_group_peek_metas  (ClutterMetaGroup *group);
 void                  _clutter_meta_group_clear_metas (ClutterMetaGroup *group);
 ClutterActorMeta *    _clutter_meta_group_get_meta    (ClutterMetaGroup *group,
                                                        const gchar      *name);
diff --git a/clutter/clutter-actor-meta.c b/clutter/clutter-actor-meta.c
index 9a6ac5e..f823268 100644
--- a/clutter/clutter-actor-meta.c
+++ b/clutter/clutter-actor-meta.c
@@ -278,7 +278,7 @@ clutter_actor_meta_set_name (ClutterActorMeta *meta,
  *
  * Since: 1.4
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_actor_meta_get_name (ClutterActorMeta *meta)
 {
   g_return_val_if_fail (CLUTTER_IS_ACTOR_META (meta), NULL);
@@ -467,7 +467,7 @@ _clutter_meta_group_remove_meta (ClutterMetaGroup *group,
  *
  * Return value: a const pointer to the #GList of #ClutterActorMeta
  */
-G_CONST_RETURN GList *
+const GList *
 _clutter_meta_group_peek_metas (ClutterMetaGroup *group)
 {
   return group->meta;
diff --git a/clutter/clutter-actor-meta.h b/clutter/clutter-actor-meta.h
index 04ef717..db5f9d4 100644
--- a/clutter/clutter-actor-meta.h
+++ b/clutter/clutter-actor-meta.h
@@ -92,7 +92,7 @@ GType clutter_actor_meta_get_type (void) G_GNUC_CONST;
 
 void                  clutter_actor_meta_set_name    (ClutterActorMeta *meta,
                                                       const gchar      *name);
-G_CONST_RETURN gchar *clutter_actor_meta_get_name    (ClutterActorMeta *meta);
+const gchar *         clutter_actor_meta_get_name    (ClutterActorMeta *meta);
 void                  clutter_actor_meta_set_enabled (ClutterActorMeta *meta,
                                                       gboolean          is_enabled);
 gboolean              clutter_actor_meta_get_enabled (ClutterActorMeta *meta);
diff --git a/clutter/clutter-actor-private.h b/clutter/clutter-actor-private.h
index b598008..b2b182e 100644
--- a/clutter/clutter-actor-private.h
+++ b/clutter/clutter-actor-private.h
@@ -157,7 +157,7 @@ gboolean           _clutter_actor_set_default_paint_volume (ClutterActor *self,
                                                             GType         check_gtype,
                                                             ClutterPaintVolume *volume);
 
-G_CONST_RETURN gchar *_clutter_actor_get_debug_name (ClutterActor *self);
+const gchar *_clutter_actor_get_debug_name (ClutterActor *self);
 
 void _clutter_actor_push_clone_paint (void);
 void _clutter_actor_pop_clone_paint  (void);
diff --git a/clutter/clutter-actor.c b/clutter/clutter-actor.c
index f113964..6d0f1b3 100644
--- a/clutter/clutter-actor.c
+++ b/clutter/clutter-actor.c
@@ -674,7 +674,7 @@ G_DEFINE_ABSTRACT_TYPE_WITH_CODE (ClutterActor,
                                   G_IMPLEMENT_INTERFACE (ATK_TYPE_IMPLEMENTOR,
                                                          atk_implementor_iface_init));
 
-G_CONST_RETURN gchar *
+const gchar *
 _clutter_actor_get_debug_name (ClutterActor *actor)
 {
   return actor->priv->name != NULL ? actor->priv->name
@@ -7158,7 +7158,7 @@ clutter_actor_set_name (ClutterActor *self,
  * Return value: the name of the actor, or %NULL. The returned string is
  *   owned by the actor and should not be modified or freed.
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_actor_get_name (ClutterActor *self)
 {
   g_return_val_if_fail (CLUTTER_IS_ACTOR (self), NULL);
diff --git a/clutter/clutter-actor.h b/clutter/clutter-actor.h
index 6b05d13..83cf3c1 100644
--- a/clutter/clutter-actor.h
+++ b/clutter/clutter-actor.h
@@ -426,7 +426,7 @@ gboolean              clutter_actor_get_paint_visibility      (ClutterActor
 
 void                  clutter_actor_set_name                  (ClutterActor          *self,
                                                                const gchar           *name);
-G_CONST_RETURN gchar *clutter_actor_get_name                  (ClutterActor          *self);
+const gchar *         clutter_actor_get_name                  (ClutterActor          *self);
 
 guint32               clutter_actor_get_gid                   (ClutterActor          *self);
 void                  clutter_actor_set_clip                  (ClutterActor          *self,
diff --git a/clutter/clutter-animator.c b/clutter/clutter-animator.c
index 0ba707c..5164a94 100644
--- a/clutter/clutter-animator.c
+++ b/clutter/clutter-animator.c
@@ -2020,7 +2020,7 @@ clutter_animator_key_get_object (const ClutterAnimatorKey *key)
  *
  * Since: 1.2
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_animator_key_get_property_name (const ClutterAnimatorKey *key)
 {
   g_return_val_if_fail (key != NULL, NULL);
diff --git a/clutter/clutter-animator.h b/clutter/clutter-animator.h
index 1f46799..bcba89f 100644
--- a/clutter/clutter-animator.h
+++ b/clutter/clutter-animator.h
@@ -161,7 +161,7 @@ void                 clutter_animator_property_set_interpolation (ClutterAnimato
 
 GType                 clutter_animator_key_get_type          (void) G_GNUC_CONST;
 GObject *             clutter_animator_key_get_object        (const ClutterAnimatorKey *key);
-G_CONST_RETURN gchar *clutter_animator_key_get_property_name (const ClutterAnimatorKey *key);
+const gchar *         clutter_animator_key_get_property_name (const ClutterAnimatorKey *key);
 GType                 clutter_animator_key_get_property_type (const ClutterAnimatorKey *key);
 gulong                clutter_animator_key_get_mode          (const ClutterAnimatorKey *key);
 gdouble               clutter_animator_key_get_progress      (const ClutterAnimatorKey *key);
diff --git a/clutter/clutter-backend.c b/clutter/clutter-backend.c
index e0a79bb..3e2f99f 100644
--- a/clutter/clutter-backend.c
+++ b/clutter/clutter-backend.c
@@ -879,7 +879,7 @@ clutter_backend_set_font_name (ClutterBackend *backend,
  *
  * Deprecated: 1.4: Use #ClutterSettings:font-name instead
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_backend_get_font_name (ClutterBackend *backend)
 {
   ClutterBackendPrivate *priv;
diff --git a/clutter/clutter-backend.h b/clutter/clutter-backend.h
index 1d18b32..5d8288f 100644
--- a/clutter/clutter-backend.h
+++ b/clutter/clutter-backend.h
@@ -71,7 +71,7 @@ void                        clutter_backend_set_double_click_distance (ClutterBa
 guint                       clutter_backend_get_double_click_distance (ClutterBackend             *backend);
 void                        clutter_backend_set_font_name             (ClutterBackend             *backend,
                                                                        const gchar                *font_name);
-G_CONST_RETURN gchar *      clutter_backend_get_font_name             (ClutterBackend             *backend);
+const gchar *      clutter_backend_get_font_name             (ClutterBackend             *backend);
 #endif /* CLUTTER_DISABLE_DEPRECATED */
 
 gdouble                     clutter_backend_get_resolution            (ClutterBackend             *backend);
diff --git a/clutter/clutter-binding-pool.c b/clutter/clutter-binding-pool.c
index 11fbf18..fa2e9a2 100644
--- a/clutter/clutter-binding-pool.c
+++ b/clutter/clutter-binding-pool.c
@@ -723,7 +723,7 @@ clutter_binding_pool_override_closure (ClutterBindingPool  *pool,
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_binding_pool_find_action (ClutterBindingPool  *pool,
                                   guint                key_val,
                                   ClutterModifierType  modifiers)
diff --git a/clutter/clutter-binding-pool.h b/clutter/clutter-binding-pool.h
index 7fc7408..77c425c 100644
--- a/clutter/clutter-binding-pool.h
+++ b/clutter/clutter-binding-pool.h
@@ -98,7 +98,7 @@ void                  clutter_binding_pool_override_closure (ClutterBindingPool
                                                              ClutterModifierType  modifiers,
                                                              GClosure            *closure);
 
-G_CONST_RETURN gchar *clutter_binding_pool_find_action      (ClutterBindingPool  *pool,
+const gchar *         clutter_binding_pool_find_action      (ClutterBindingPool  *pool,
                                                              guint                key_val,
                                                              ClutterModifierType  modifiers);
 void                  clutter_binding_pool_remove_action    (ClutterBindingPool  *pool,
diff --git a/clutter/clutter-color.c b/clutter/clutter-color.c
index 3944c89..e8ecf44 100644
--- a/clutter/clutter-color.c
+++ b/clutter/clutter-color.c
@@ -115,7 +115,7 @@ static const ClutterColor const static_colors[] = {
  *
  * Since: 1.6
  */
-G_CONST_RETURN ClutterColor *
+const ClutterColor *
 clutter_color_get_static (ClutterStaticColor color)
 {
   g_return_val_if_fail (color >= CLUTTER_COLOR_WHITE &&
@@ -1055,7 +1055,7 @@ clutter_value_set_color (GValue             *value,
  *
  * Since: 0.8.4
  */
-G_CONST_RETURN ClutterColor *
+const ClutterColor *
 clutter_value_get_color (const GValue *value)
 {
   g_return_val_if_fail (CLUTTER_VALUE_HOLDS_COLOR (value), NULL);
diff --git a/clutter/clutter-color.h b/clutter/clutter-color.h
index 3649ce8..f36878d 100644
--- a/clutter/clutter-color.h
+++ b/clutter/clutter-color.h
@@ -140,7 +140,7 @@ struct _ClutterParamSpecColor
 
 void                         clutter_value_set_color (GValue             *value,
                                                       const ClutterColor *color);
-G_CONST_RETURN ClutterColor *clutter_value_get_color (const GValue       *value);
+const ClutterColor          *clutter_value_get_color (const GValue       *value);
 
 GType       clutter_param_color_get_type (void) G_GNUC_CONST;
 GParamSpec *clutter_param_spec_color     (const gchar        *name,
@@ -254,7 +254,7 @@ typedef enum { /*< prefix=CLUTTER_COLOR >*/
   CLUTTER_COLOR_TRANSPARENT
 } ClutterStaticColor;
 
-G_CONST_RETURN ClutterColor *clutter_color_get_static (ClutterStaticColor color);
+const ClutterColor *clutter_color_get_static (ClutterStaticColor color);
 
 G_END_DECLS
 
diff --git a/clutter/clutter-event.c b/clutter/clutter-event.c
index 1b0f563..ddb075b 100644
--- a/clutter/clutter-event.c
+++ b/clutter/clutter-event.c
@@ -943,7 +943,7 @@ clutter_get_current_event_time (void)
  *
  * Since: 1.2
  */
-G_CONST_RETURN ClutterEvent *
+const ClutterEvent *
 clutter_get_current_event (void)
 {
   ClutterMainContext *context = _clutter_context_get_default ();
diff --git a/clutter/clutter-event.h b/clutter/clutter-event.h
index ebaa880..a6d5e10 100644
--- a/clutter/clutter-event.h
+++ b/clutter/clutter-event.h
@@ -430,7 +430,7 @@ void                   clutter_event_set_device             (ClutterEvent
 guint32                clutter_keysym_to_unicode            (guint               keyval);
 
 guint32                clutter_get_current_event_time       (void);
-G_CONST_RETURN ClutterEvent *clutter_get_current_event      (void);
+const ClutterEvent *   clutter_get_current_event            (void);
 
 G_END_DECLS
 
diff --git a/clutter/clutter-input-device.c b/clutter/clutter-input-device.c
index 3f44b97..135a487 100644
--- a/clutter/clutter-input-device.c
+++ b/clutter/clutter-input-device.c
@@ -785,7 +785,7 @@ clutter_input_device_get_pointer_stage (ClutterInputDevice *device)
  *
  * Since: 1.2
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_input_device_get_device_name (ClutterInputDevice *device)
 {
   g_return_val_if_fail (CLUTTER_IS_INPUT_DEVICE (device), NULL);
diff --git a/clutter/clutter-input-device.h b/clutter/clutter-input-device.h
index 6772a39..7b3a75b 100644
--- a/clutter/clutter-input-device.h
+++ b/clutter/clutter-input-device.h
@@ -135,7 +135,7 @@ void                    clutter_input_device_get_device_coords  (ClutterInputDev
                                                                  gint                *y);
 ClutterActor *          clutter_input_device_get_pointer_actor  (ClutterInputDevice  *device);
 ClutterStage *          clutter_input_device_get_pointer_stage  (ClutterInputDevice  *device);
-G_CONST_RETURN gchar *  clutter_input_device_get_device_name    (ClutterInputDevice  *device);
+const gchar *           clutter_input_device_get_device_name    (ClutterInputDevice  *device);
 ClutterInputMode        clutter_input_device_get_device_mode    (ClutterInputDevice  *device);
 gboolean                clutter_input_device_get_has_cursor     (ClutterInputDevice  *device);
 void                    clutter_input_device_set_enabled        (ClutterInputDevice  *device,
diff --git a/clutter/clutter-interval.c b/clutter/clutter-interval.c
index 9d14601..2ae3c1c 100644
--- a/clutter/clutter-interval.c
+++ b/clutter/clutter-interval.c
@@ -918,7 +918,7 @@ clutter_interval_compute_value (ClutterInterval *interval,
  *
  * Since: 1.4
  */
-G_CONST_RETURN GValue *
+const GValue *
 clutter_interval_compute (ClutterInterval *interval,
                           gdouble          factor)
 {
diff --git a/clutter/clutter-interval.h b/clutter/clutter-interval.h
index 9346c72..0cc6461 100644
--- a/clutter/clutter-interval.h
+++ b/clutter/clutter-interval.h
@@ -154,7 +154,7 @@ gboolean         clutter_interval_compute_value      (ClutterInterval *interval,
                                                       gdouble          factor,
                                                       GValue          *value);
 
-G_CONST_RETURN GValue *clutter_interval_compute      (ClutterInterval *interval,
+const GValue *clutter_interval_compute      (ClutterInterval *interval,
                                                       gdouble          factor);
 
 void clutter_interval_register_progress_func (GType               value_type,
diff --git a/clutter/clutter-model.c b/clutter/clutter-model.c
index 880b73a..644e55e 100644
--- a/clutter/clutter-model.c
+++ b/clutter/clutter-model.c
@@ -1382,7 +1382,7 @@ clutter_model_remove (ClutterModel *model,
  *
  * Since: 0.6
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_model_get_column_name (ClutterModel *model,
                                guint         column)
 {
diff --git a/clutter/clutter-model.h b/clutter/clutter-model.h
index a49ddbb..a8217ba 100644
--- a/clutter/clutter-model.h
+++ b/clutter/clutter-model.h
@@ -226,7 +226,7 @@ void                  clutter_model_remove             (ClutterModel     *model,
 
 guint                 clutter_model_get_n_rows         (ClutterModel     *model);
 guint                 clutter_model_get_n_columns      (ClutterModel     *model);
-G_CONST_RETURN gchar *clutter_model_get_column_name    (ClutterModel     *model,
+const gchar *         clutter_model_get_column_name    (ClutterModel     *model,
                                                         guint             column);
 GType                 clutter_model_get_column_type    (ClutterModel     *model,
                                                         guint             column);
diff --git a/clutter/clutter-private.h b/clutter/clutter-private.h
index ed64ae4..9939997 100644
--- a/clutter/clutter-private.h
+++ b/clutter/clutter-private.h
@@ -195,7 +195,7 @@ ClutterActor *          _clutter_context_peek_shader_stack      (void);
 guint32                 _clutter_context_acquire_id             (gpointer      key);
 void                    _clutter_context_release_id             (guint32       id_);
 
-G_CONST_RETURN gchar *_clutter_gettext (const gchar *str);
+const gchar *_clutter_gettext (const gchar *str);
 
 gboolean      _clutter_feature_init (GError **error);
 
diff --git a/clutter/clutter-script.c b/clutter/clutter-script.c
index 5ba91d6..2c9e5af 100644
--- a/clutter/clutter-script.c
+++ b/clutter/clutter-script.c
@@ -799,7 +799,7 @@ clutter_script_get_type_from_name (ClutterScript *script,
  *
  * Since: 0.6
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_get_script_id (GObject *gobject)
 {
   g_return_val_if_fail (G_IS_OBJECT (gobject), NULL);
diff --git a/clutter/clutter-script.h b/clutter/clutter-script.h
index 2b9fe79..58702dd 100644
--- a/clutter/clutter-script.h
+++ b/clutter/clutter-script.h
@@ -168,7 +168,7 @@ void           clutter_script_ensure_objects       (ClutterScript  *script);
 GType          clutter_script_get_type_from_name   (ClutterScript  *script,
                                                     const gchar    *type_name);
 
-G_CONST_RETURN gchar *clutter_get_script_id        (GObject        *gobject);
+const gchar *  clutter_get_script_id               (GObject        *gobject);
 
 void           clutter_script_connect_signals      (ClutterScript  *script,
                                                     gpointer        user_data);
diff --git a/clutter/clutter-scriptable.c b/clutter/clutter-scriptable.c
index 501f1df..e4ffdc4 100644
--- a/clutter/clutter-scriptable.c
+++ b/clutter/clutter-scriptable.c
@@ -104,7 +104,7 @@ clutter_scriptable_set_id (ClutterScriptable *scriptable,
  *
  * Since: 0.6
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_scriptable_get_id (ClutterScriptable *scriptable)
 {
   ClutterScriptableIface *iface;
diff --git a/clutter/clutter-scriptable.h b/clutter/clutter-scriptable.h
index e2c5ff1..f47aac4 100644
--- a/clutter/clutter-scriptable.h
+++ b/clutter/clutter-scriptable.h
@@ -91,7 +91,7 @@ GType                 clutter_scriptable_get_type            (void) G_GNUC_CONST
 
 void                  clutter_scriptable_set_id              (ClutterScriptable *scriptable,
                                                               const gchar       *id_);
-G_CONST_RETURN gchar *clutter_scriptable_get_id              (ClutterScriptable *scriptable);
+const gchar *         clutter_scriptable_get_id              (ClutterScriptable *scriptable);
 gboolean              clutter_scriptable_parse_custom_node   (ClutterScriptable *scriptable,
                                                               ClutterScript     *script,
                                                               GValue            *value,
diff --git a/clutter/clutter-shader-types.c b/clutter/clutter-shader-types.c
index 9ed3998..f18e344 100644
--- a/clutter/clutter-shader-types.c
+++ b/clutter/clutter-shader-types.c
@@ -485,7 +485,7 @@ clutter_value_set_shader_matrix (GValue       *value,
  *
  * Since: 0.8
  */
-G_CONST_RETURN gfloat *
+const gfloat *
 clutter_value_get_shader_float (const GValue *value,
                                 gsize        *length)
 {
@@ -517,7 +517,7 @@ clutter_value_get_shader_float (const GValue *value,
  *
  * Since: 0.8
  */
-G_CONST_RETURN gint *
+const gint *
 clutter_value_get_shader_int (const GValue *value,
                               gsize        *length)
 {
@@ -549,7 +549,7 @@ clutter_value_get_shader_int (const GValue *value,
  *
  * Since: 0.8
  */
-G_CONST_RETURN gfloat *
+const gfloat *
 clutter_value_get_shader_matrix (const GValue *value,
                                  gsize        *length)
 {
diff --git a/clutter/clutter-shader-types.h b/clutter/clutter-shader-types.h
index 1b48426..c5d30d1 100644
--- a/clutter/clutter-shader-types.h
+++ b/clutter/clutter-shader-types.h
@@ -84,11 +84,11 @@ void                    clutter_value_set_shader_int    (GValue       *value,
 void                    clutter_value_set_shader_matrix (GValue       *value,
                                                          gint          size,
                                                          const gfloat *matrix);
-G_CONST_RETURN gfloat * clutter_value_get_shader_float  (const GValue *value,
+const gfloat *          clutter_value_get_shader_float  (const GValue *value,
                                                          gsize        *length);
-G_CONST_RETURN gint   * clutter_value_get_shader_int    (const GValue *value,
+const gint   *          clutter_value_get_shader_int    (const GValue *value,
                                                          gsize        *length);
-G_CONST_RETURN gfloat * clutter_value_get_shader_matrix (const GValue *value,
+const gfloat *          clutter_value_get_shader_matrix (const GValue *value,
                                                          gsize        *length);
 
 G_END_DECLS
diff --git a/clutter/clutter-shader.c b/clutter/clutter-shader.c
index e2109c7..d7a9867 100644
--- a/clutter/clutter-shader.c
+++ b/clutter/clutter-shader.c
@@ -815,7 +815,7 @@ _clutter_shader_release_all (void)
  *
  * Since: 0.6
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_shader_get_fragment_source (ClutterShader *shader)
 {
   g_return_val_if_fail (CLUTTER_IS_SHADER (shader), NULL);
@@ -835,7 +835,7 @@ clutter_shader_get_fragment_source (ClutterShader *shader)
  *
  * Since: 0.6
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_shader_get_vertex_source (ClutterShader *shader)
 {
   g_return_val_if_fail (CLUTTER_IS_SHADER (shader), NULL);
diff --git a/clutter/clutter-shader.h b/clutter/clutter-shader.h
index ef72192..10011a7 100644
--- a/clutter/clutter-shader.h
+++ b/clutter/clutter-shader.h
@@ -117,8 +117,8 @@ void                  clutter_shader_set_fragment_source      (ClutterShader
                                                                const gchar        *data,
                                                                gssize              length);
 
-G_CONST_RETURN gchar *clutter_shader_get_vertex_source        (ClutterShader      *shader);
-G_CONST_RETURN gchar *clutter_shader_get_fragment_source      (ClutterShader      *shader);
+const gchar *clutter_shader_get_vertex_source        (ClutterShader      *shader);
+const gchar *clutter_shader_get_fragment_source      (ClutterShader      *shader);
 
 void                  clutter_shader_set_uniform              (ClutterShader      *shader,
                                                                const gchar        *name,
diff --git a/clutter/clutter-stage.c b/clutter/clutter-stage.c
index de31aef..bc388b5 100644
--- a/clutter/clutter-stage.c
+++ b/clutter/clutter-stage.c
@@ -2367,7 +2367,7 @@ clutter_stage_set_title (ClutterStage       *stage,
  *
  * Since: 0.4
  **/
-G_CONST_RETURN gchar *
+const gchar *
 clutter_stage_get_title (ClutterStage       *stage)
 {
   g_return_val_if_fail (CLUTTER_IS_STAGE (stage), NULL);
diff --git a/clutter/clutter-stage.h b/clutter/clutter-stage.h
index e9c8681..3098c79 100644
--- a/clutter/clutter-stage.h
+++ b/clutter/clutter-stage.h
@@ -218,7 +218,7 @@ gboolean      clutter_stage_event             (ClutterStage       *stage,
 
 void                  clutter_stage_set_title          (ClutterStage *stage,
                                                         const gchar  *title);
-G_CONST_RETURN gchar *clutter_stage_get_title          (ClutterStage *stage);
+const gchar *         clutter_stage_get_title          (ClutterStage *stage);
 void                  clutter_stage_set_user_resizable (ClutterStage *stage,
 						        gboolean      resizable);
 gboolean              clutter_stage_get_user_resizable (ClutterStage *stage);
diff --git a/clutter/clutter-state.c b/clutter/clutter-state.c
index fb875e3..0f38668 100644
--- a/clutter/clutter-state.c
+++ b/clutter/clutter-state.c
@@ -1730,7 +1730,7 @@ clutter_state_key_get_object (const ClutterStateKey *state_key)
  *
  * Since: 1.4
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_state_key_get_property_name (const ClutterStateKey *state_key)
 {
   g_return_val_if_fail (state_key, NULL);
@@ -1751,7 +1751,7 @@ clutter_state_key_get_property_name (const ClutterStateKey *state_key)
  *
  * Since: 1.4
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_state_key_get_source_state_name (const ClutterStateKey *state_key)
 {
   g_return_val_if_fail (state_key, NULL);
@@ -1775,7 +1775,7 @@ clutter_state_key_get_source_state_name (const ClutterStateKey *state_key)
  *
  * Since: 1.4
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_state_key_get_target_state_name (const ClutterStateKey *state_key)
 {
   g_return_val_if_fail (state_key, NULL);
@@ -1934,7 +1934,7 @@ clutter_state_get_duration (ClutterState *state,
  *
  * Since: 1.4
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_state_get_state (ClutterState *state)
 {
   g_return_val_if_fail (CLUTTER_IS_STATE (state), NULL);
diff --git a/clutter/clutter-state.h b/clutter/clutter-state.h
index e885a40..79342c7 100644
--- a/clutter/clutter-state.h
+++ b/clutter/clutter-state.h
@@ -138,7 +138,7 @@ void              clutter_state_set_animator  (ClutterState    *state,
 ClutterAnimator * clutter_state_get_animator  (ClutterState    *state,
                                                const gchar     *source_state_name,
                                                const gchar     *target_state_name);
-G_CONST_RETURN gchar *clutter_state_get_state (ClutterState    *state);
+const gchar *     clutter_state_get_state     (ClutterState    *state);
 
 /*
  * ClutterStateKey
@@ -152,9 +152,9 @@ gboolean              clutter_state_key_get_value             (const ClutterStat
                                                                GValue                *value);
 GType                 clutter_state_key_get_property_type     (const ClutterStateKey *key);
 GObject *             clutter_state_key_get_object            (const ClutterStateKey *state_key);
-G_CONST_RETURN gchar *clutter_state_key_get_property_name     (const ClutterStateKey *state_key);
-G_CONST_RETURN gchar *clutter_state_key_get_source_state_name (const ClutterStateKey *state_key);
-G_CONST_RETURN gchar *clutter_state_key_get_target_state_name (const ClutterStateKey *state_key);
+const gchar *         clutter_state_key_get_property_name     (const ClutterStateKey *state_key);
+const gchar *         clutter_state_key_get_source_state_name (const ClutterStateKey *state_key);
+const gchar *         clutter_state_key_get_target_state_name (const ClutterStateKey *state_key);
 
 G_END_DECLS
 
diff --git a/clutter/clutter-text.c b/clutter/clutter-text.c
index 4566cc3..744b294 100644
--- a/clutter/clutter-text.c
+++ b/clutter/clutter-text.c
@@ -4035,7 +4035,7 @@ clutter_text_get_font_description (ClutterText *self)
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_text_get_font_name (ClutterText *text)
 {
   g_return_val_if_fail (CLUTTER_IS_TEXT (text), NULL);
@@ -4140,7 +4140,7 @@ out:
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 clutter_text_get_text (ClutterText *self)
 {
   g_return_val_if_fail (CLUTTER_IS_TEXT (self), NULL);
diff --git a/clutter/clutter-text.h b/clutter/clutter-text.h
index 9edbbca..7430cc2 100644
--- a/clutter/clutter-text.h
+++ b/clutter/clutter-text.h
@@ -103,7 +103,7 @@ ClutterActor *        clutter_text_new_full             (const gchar          *f
 ClutterActor *        clutter_text_new_with_text        (const gchar          *font_name,
                                                          const gchar          *text);
 
-G_CONST_RETURN gchar *clutter_text_get_text             (ClutterText          *self);
+const gchar *         clutter_text_get_text             (ClutterText          *self);
 void                  clutter_text_set_text             (ClutterText          *self,
                                                          const gchar          *text);
 void                  clutter_text_set_markup           (ClutterText          *self,
@@ -114,7 +114,7 @@ void                  clutter_text_get_color            (ClutterText          *s
                                                          ClutterColor         *color);
 void                  clutter_text_set_font_name        (ClutterText          *self,
                                                          const gchar          *font_name);
-G_CONST_RETURN gchar *clutter_text_get_font_name        (ClutterText          *self);
+const gchar          *clutter_text_get_font_name        (ClutterText          *self);
 void                  clutter_text_set_font_description (ClutterText          *self,
                                                          PangoFontDescription *font_desc);
 PangoFontDescription *clutter_text_get_font_description (ClutterText          *self);
diff --git a/clutter/clutter-units.c b/clutter/clutter-units.c
index 422840a..808f830 100644
--- a/clutter/clutter-units.c
+++ b/clutter/clutter-units.c
@@ -773,7 +773,7 @@ clutter_value_set_units (GValue             *value,
  *
  * Since: 0.8
  */
-G_CONST_RETURN ClutterUnits *
+const ClutterUnits *
 clutter_value_get_units (const GValue *value)
 {
   g_return_val_if_fail (CLUTTER_VALUE_HOLDS_UNITS (value), NULL);
diff --git a/clutter/clutter-units.h b/clutter/clutter-units.h
index c75410c..8a29232 100644
--- a/clutter/clutter-units.h
+++ b/clutter/clutter-units.h
@@ -181,7 +181,7 @@ GParamSpec *                 clutter_param_spec_units (const gchar        *name,
 
 void                         clutter_value_set_units  (GValue             *value,
                                                        const ClutterUnits *units);
-G_CONST_RETURN ClutterUnits *clutter_value_get_units  (const GValue       *value);
+const ClutterUnits          *clutter_value_get_units  (const GValue       *value);
 
 G_END_DECLS
 
diff --git a/clutter/clutter-util.c b/clutter/clutter-util.c
index aa4056d..3713a24 100644
--- a/clutter/clutter-util.c
+++ b/clutter/clutter-util.c
@@ -68,7 +68,7 @@ clutter_util_next_p2 (gint a)
  *
  * Return value: the translated string
  */
-G_CONST_RETURN gchar *
+const gchar *
 _clutter_gettext (const gchar *str)
 {
   return g_dgettext (GETTEXT_PACKAGE, str);
diff --git a/clutter/glx/clutter-backend-glx.c b/clutter/glx/clutter-backend-glx.c
index 684ad29..18c3232 100644
--- a/clutter/glx/clutter-backend-glx.c
+++ b/clutter/glx/clutter-backend-glx.c
@@ -59,7 +59,7 @@ static ClutterBackendGLX *backend_singleton = NULL;
 
 static gchar    *clutter_vblank_name = NULL;
 
-G_CONST_RETURN gchar*
+const gchar*
 _clutter_backend_glx_get_vblank_method (void)
 {
   return clutter_vblank_name;
diff --git a/clutter/x11/clutter-backend-x11.c b/clutter/x11/clutter-backend-x11.c
index c029500..e26c4a6 100644
--- a/clutter/x11/clutter-backend-x11.c
+++ b/clutter/x11/clutter-backend-x11.c
@@ -1028,7 +1028,7 @@ clutter_x11_remove_filter (ClutterX11FilterFunc func,
  *   pointer to the internal list of input devices; the returned list is
  *   owned by Clutter and should not be modified or freed
  */
-G_CONST_RETURN GSList *
+const GSList *
 clutter_x11_get_input_devices (void)
 {
   ClutterDeviceManager *manager;
diff --git a/clutter/x11/clutter-x11.h b/clutter/x11/clutter-x11.h
index 66611bd..262a546 100644
--- a/clutter/x11/clutter-x11.h
+++ b/clutter/x11/clutter-x11.h
@@ -127,7 +127,7 @@ gboolean clutter_x11_has_event_retrieval (void);
 ClutterStage *clutter_x11_get_stage_from_window (Window win);
 
 #ifndef CLUTTER_DISABLE_DEPRECATED
-G_CONST_RETURN GSList* clutter_x11_get_input_devices (void) G_GNUC_DEPRECATED;
+const GSList* clutter_x11_get_input_devices (void) G_GNUC_DEPRECATED;
 #endif
 
 void     clutter_x11_enable_xinput (void);
--
cgit v0.9
